mechaclient.py

```diff
@@ -139,3 +141,32 @@ def api_handler(self) -> object:
         Mecha's API connection
         """
         return self._api_handler
+
+    @property
+    def board(self) -> RatBoard:
+        """
+        Mecha's RatBoard Object
+
+        """
+        if not self._rat_board:
+            self._rat_board = RatBoard()  # Create Rat Board Object
+        return self._rat_board
+
+    @board.setter
+    def board(self, value):
+        """
+        Mecha's RatBoard Setter
+        """
+        if not isinstance(value, RatBoard):
+            raise TypeError("board property must be of type RatBoard.")
+
+        log.warning("Board Setter invoked!")
+        self._rat_board = value
+
+    @board.deleter
+    def board(self):
+        """
+        Mecha's Ratboard Deleter
+        """
+        log.warning("Board Deleted!")
+        del self._rat_board
```

Comment:
once the underlying is freed, methings we need to assign `None` to the attribute to prevent an attribute error in the getter. Have you tested this use-case yet?

Reply:
This is almost a direct copy paste from Fact Manager's attribute.  Do you want to change that as well, or is this a different case?

Reply:
yeah, I think both cases have this particular bug.  before we make a code change, lets run a test
```py
def test_rat_board_deleter(bot_fx):
   """
   verifies deleting the rat board property doesn't raise an attribute error on next access
   """
   old_board = bot_fx.board
   del bot_fx.board
   new_board = bot_fx.board   # this line i expect to raise an AttributeError
   assert new_board is not old_board  
```